lib/orbit/ocs/stream/pipeline.ex

```diff
@@ -96,4 +114,87 @@ defmodule Orbit.Ocs.Stream.Pipeline do
     # This is required per the framework, but could be useful down the line
     # to track which messages actually make it all the way into state.
   end
+
+  # Kinesis Stream Persistence
+
+  defp get_producer_opts(opts) do
+    resume_position = load_resume_position()
+    opts ++ [state: %{resume_position: resume_position}]
+  end
+
+  @spec load_resume_position() :: BroadwayKinesis.SubscribeToShard.starting_position()
+  defp load_resume_position do
+    cutoff = start_of_service_date()
+
+    with stream_name when not is_nil(stream_name) <- get_stream_name(),
+         stream_state when not is_nil(stream_state) <- load_stream_state(stream_name),
+         false <- expired?(stream_state, cutoff) do
+      Logger.info("Will resume OCS Kinesis stream from #{stream_state.resume_position}")
+
+      {:after_sequence_number, stream_state.resume_position}
+    else
+      _ ->
+        Logger.info("Will begin OCS Kinesis stream from timestamp #{cutoff}")
+        {:at_timestamp, cutoff}
+    end
+  end
+
+  @spec load_stream_state(String.t()) :: KinesisStreamState.t() | nil
+  defp load_stream_state(stream_name) do
+    stream_state = Repo.get_by(KinesisStreamState, stream_name: stream_name)
+
+    if stream_state == nil do
+      Logger.warning("No prior state saved for OCS Kinesis stream #{stream_name}")
+    end
+
+    stream_state
+  end
+
+  @spec expired?(KinesisStreamState.t(), DateTime.t()) :: boolean()
+  defp expired?(%KinesisStreamState{last_message_timestamp: timestamp}, cutoff) do
+    expired = :gt == DateTime.compare(start_of_service_date(), timestamp)
+
+    if expired do
+      Logger.warning("Last OCS Kinesis timestamp #{timestamp} is older than cutoff #{cutoff}.")
+    end
+
+    expired
+  end
+
+  defp start_of_service_date do
```

Comment:
suggestion: this name slightly suggests the cutoff time on a particular service date, perhaps `midnight_on_service_date` would be better?